application/helpers/admin/statistics_helper.php

```diff
@@ -2535,17 +2537,58 @@ protected function displayResults($outputs, $results, $rt, $outputType, $surveyi
                 //answer text
                 $fname = "$al[1] ($al[0])";
             }
-             //are there some results to play with?
-            if ($results > 0) {
+            if (($outputs['qtype'] != Question::QT_M_MULTIPLE_CHOICE) and ($outputs['qtype'] != Question::QT_P_MULTIPLE_CHOICE_WITH_COMMENTS)) {
+                //are there some results to play with?
+                if ($results > 0) {
+                    //calculate percentage
+                    $gdata[] = ($row / $results) * 100;
+                } else {
+                    //no data!
+                    $gdata[] = "N/A";
+                }
+            } else { // For multiple choice : we need total answered or viewed on all columns
+                if ($totalMultiAnswers !== false) {
+                    $totalMultiAnswers = $results; // By default all answers
+                    if ($noncompleted) {
+                        $criteria = new CDbCriteria();
```

Comment:
Maybe we can wrap the above into a function for the above? I know this file and function is already big, but maybe if we start breaking it into smaller pieces it would be more testable and more understandable.